pkg/auth/types.go

```diff
@@ -0,0 +1,61 @@
+package auth
+
+import (
+	"github.com/docker/docker-credential-helpers/credentials"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+const (
+	remoteCredentialsPrefix = "docker-credential-"
+	tokenUsername           = "<token>"
+)
+
+// CredentialConf represents the configuration
+// saved in credentials Store
+type CredentialConf struct {
+	Username      string
+	Password      string
+	ServerAddress string
+	// IdentityToken is used to authenticate the user and get
+	// an access token for the registry.
+	IdentityToken string `json:"identitytoken,omitempty"`
+	// RegistryToken is a bearer token to be sent to a registry
+	RegistryToken string `json:"registrytoken,omitempty"`
+}
+
+// NewCredentialConfFromDockerCreds creates a new CredentialConf from the docker-cli credentials
+func NewCredentialConfFromDockerCreds(dockerCreds *credentials.Credentials, serverAddress string) *CredentialConf {
+	var credsConf CredentialConf
+	if dockerCreds.Username == tokenUsername {
+		credsConf.IdentityToken = dockerCreds.Secret
+	} else {
+		credsConf.Password = dockerCreds.Secret
+		credsConf.Username = dockerCreds.Username
+	}
+
+	credsConf.ServerAddress = serverAddress
+	return &credsConf
+}
+
+// NewAuthConfig creates a new auth.Credential from a credential configuration
+func NewAuthConfig(creds *CredentialConf) *auth.Credential {
+	return &auth.Credential{
+		Username:     creds.Username,
+		Password:     creds.Password,
+		AccessToken:  creds.RegistryToken,
+		RefreshToken: creds.IdentityToken,
+	}
+}
```

Comment:
Once we remove `CredentialConf`, we don't need this method anymore.

Reply:
yeah, that looks much better